src/main/java/org/betonquest/betonquest/events/FolderEvent.java

```diff
@@ -87,12 +104,15 @@ public void run() {
                 BetonQuest.event(profile, event);
             }
             if (!chosenList.isEmpty()) {
+                register();
                 new BukkitRunnable() {
                     @Override
                     public void run() {
                         final EventID event = chosenList.remove(0);
-                        if (chosenList.isEmpty()) {
+                        if (event == null || cancelled.remove(profile.getProfileUUID())) {
```

Comment:
The change from `chosenList.isEmpty()` (after removing the last element) to `event == null` (when the folder event is run the next time) will cause a `NoSuchElementException` as `0` will be out of bounds of an empty `List`.

You cound instead use this method from the `Deque` interface: `Deque.pollFirst()` (but requires to change the type of `chosenList` to `Deque` and initializing it with `e.g. a `LinkedList` instead).